_rules/inline-link-in-paragraph-has-distinguishable-border-36f116.md

```diff
@@ -0,0 +1,187 @@
+- **(with other elements)** which has [inline boxes][inline box] generated from an element containing [visible][] [text nodes][text node], 
+- **(that are not links)** which do not have a [semantic link][] as a [descendant][] in the [flat tree][].
+
+**Note:** The test targets are visible link elements that are part of a block of content that is displayed inline.
+
+## Expectation
+
+Each target element has a [visible][] [border](https://drafts.csswg.org/css-backgrounds/#borders).
+
+## Assumptions
+
+- Any change in font is sufficiently distinguishable, and fonts are loaded when they are present.
+- If multiple colors are used in the visible text nodes of the block of content then color can not be a distinguishing factor.
+- If `border` is used in the different visible text nodes of the block of content then `border` can not be a distinguishing factor.
+
+## Accessibility Support
+
+_No accessibility support issues known._
+
+## Background
+
+- [Understanding Success Criterion 1.4.1: Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
+- [G183: Using a contrast ratio of 3:1 with surrounding text and providing additional visual cues on focus for links or controls where color alone is used to identify them](https://www.w3.org/WAI/WCAG21/Techniques/general/G183)
+- [F73: Failure of Success Criterion 1.4.1 due to creating links that are not visually evident without color vision](https://www.w3.org/WAI/WCAG21/Techniques/failures/F73)
+- [C15: Using CSS to change the presentation of a user interface component when it receives focus](https://www.w3.org/WAI/WCAG21/Techniques/css/C15)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This link, that is a descendant of a paragraph element, has a distinguishing bottom border in addition to being distinguishable by color.
+
+```html
+<style>
+	p {
+		color: black;
+		background-color: white;
+	}
+	a.test {
+		color: blue;
+		background-color: white;
+		text-decoration: none;
+		border-style: solid;
+		border-color: red;
+		border-width: 0px;
+		border-bottom-width: 1px;
+	}
+</style>
+<p>Read about WAI on the <a class="test" href="http://w3.org/WAI">WAI webpage</a>.</p>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This link, that is a descendant of a paragraph element, has no visual cues of being recognized as a link with the underline removed.
+
+```html
+<style>
+	a.test {
+		text-decoration: none;
+	}
+</style>
+<p>Read about WAI on the <a class="test" href="http://w3.org/WAI">WAI webpage</a>.</p>
+```
+
+#### Failed Example 2
+
+This link, that is a descendant of a paragraph element, has a not visible border because it has a width of zero.
+
+```html
+<style>
+	* {
+		color: black;
+		background-color: white;
+	}
+	a.test {
+		text-decoration: none;
+		border-style: solid;
+		border-color: blue;
+		border-width: 0px;
+	}
+</style>
+<p>Read about WAI on the <a class="test" href="http://w3.org/WAI">WAI webpage</a>.</p>
```

Comment:
This example has black link text next to black non-link text. This doesn't make it clear that this fails because of the lack of a border and doesn't make the link distinguishable with colour alone, because it isn't distinguishable at all!